src/Sylius/Component/Core/Repository/ProductRepositoryInterface.php

```diff
@@ -52,14 +52,20 @@ public function findOneByIdAndChannel($id, ChannelInterface $channel = null);
     public function findOneBySlugAndChannel($slug, ChannelInterface $channel);
 
     /**
+     * @param string $slug
+     *
+     * @return ProductInterface|null
+     */
+    public function findOneBySlug($slug);
```

Comment:
On Shop we should always use `findOneBySlugAndChannel` to be sure the product is enabled for given channel.